@here/harp-datasource-protocol/lib/Expr.ts

```diff
@@ -449,6 +559,33 @@ export class CaseExpr extends Expr {
     }
 }
 
+/**
+ * Expressions that represent literal/immediate values.
+ *
+ * @hidden
+ */
+export type LiteralExpr =
+    | StringLiteralExpr
+    | NumberLiteralExpr
+    | BooleanLiteralExpr
+    | ObjectLiteralExpr
+    | NullLiteralExpr;
+
+/**
+ * Check if `expr` represents literal/immediate values i.e conforms to [[LiteralExpr]].
+ *
+ * @hidden
+ */
+export function isLiteralExpr(expr: Expr): expr is LiteralExpr {
+    return (
```

Comment:
please replace the isLiteralExpr with just an instanceof LiteralExpr when if you add the abstract class LiteralExpr

Reply:
Not needed now, if we have one base class.